src/service_info.rs

```diff
@@ -1060,6 +1072,16 @@ pub(crate) fn split_sub_domain(domain: &str) -> (&str, Option<&str>) {
     }
 }
 
+#[non_exhaustive]
+pub struct ResolvedService {
```

Comment:
One edge case is:  if `addresses` is empty in this struct, then it would not be `resolved`.  Should we call this struct just `Service` instead?

Reply:
If we have decided to split among client and server, we can return an `Option` like we are doing right now, indeed Option<ServiceInfo>` can have empty `addresses`. Therefore, `Option<ResolvedService>` could it be a solution?